cedar-policy-core/src/parser/cst_to_ast.rs

```diff
@@ -4645,6 +4662,26 @@ mod tests {
         expect_reserved_ident("foo::is::bar::principal", "is");
     }
 
+    #[test]
+    fn reserved_namespace() {
+        assert_matches!(parse_expr(r#"__cedar::"""#),
+            Err(errs) if matches!(errs.as_ref().first(),
+                ParseError::ToAST(to_ast_err) if matches!(to_ast_err.kind(),
+                    ToASTErrorKind::ReservedNamespace(n) if *n == "__cedar".parse::<Name>().unwrap())));
+        assert_matches!(parse_expr(r#"__cedar::A::"""#),
+            Err(errs) if matches!(errs.as_ref().first(),
+                ParseError::ToAST(to_ast_err) if matches!(to_ast_err.kind(),
+                    ToASTErrorKind::ReservedNamespace(n) if *n == "__cedar::A".parse::<Name>().unwrap())));
+        assert_matches!(parse_expr(r#"[A::"", __cedar::Action::"action"]"#),
+            Err(errs) if matches!(errs.as_ref().first(),
+                ParseError::ToAST(to_ast_err) if matches!(to_ast_err.kind(),
+                    ToASTErrorKind::ReservedNamespace(n) if *n == "__cedar::Action".parse::<Name>().unwrap())));
+        assert_matches!(parse_expr(r#"principal is __cedar::A"#),
+            Err(errs) if matches!(errs.as_ref().first(),
+                ParseError::ToAST(to_ast_err) if matches!(to_ast_err.kind(),
+                    ToASTErrorKind::ReservedNamespace(n) if *n == "__cedar::A".parse::<Name>().unwrap())));
```

Comment:
These tests are great; can we also get tests for some additional cases?
- `Foo::__cedar::""`
- `Foo::__cedar::A::""`